packages/operator/pkg/service/batch/job_validate.go

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ *     Copyright 2021 EPAM Systems
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package batch
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	api_types "github.com/odahu/odahu-flow/packages/operator/pkg/apis/batch"
+	odahu_errors "github.com/odahu/odahu-flow/packages/operator/pkg/errors"
+)
+
+const (
+	EmptyServiceJobField = `field "%s"" must be set for BatchInferenceJob 
+or as a default for a corresponding BatchInferenceService`
+	ConnectionNotFound = `connection from "%s" with name "%s" is not found`
+)
+
+func DefaultJob(job *api_types.InferenceJob, service api_types.InferenceService) {
+
+	job.ID = "job-" + service.ID + "-" + uuid.New().String()[:5]
+
+	if len(job.Spec.BatchRequestID) == 0 {
+		job.Spec.BatchRequestID = uuid.New().String()
+	}
+
+	if job.Spec.DataSource == nil {
+		job.Spec.DataSource = service.Spec.DataSource
+	}
+	if job.Spec.OutputDestination == nil {
+		job.Spec.OutputDestination = service.Spec.OutputDestination
+	}
+	if job.Spec.NodeSelector == nil {
+		job.Spec.NodeSelector = service.Spec.NodeSelector
+	}
+	if job.Spec.Resources == nil {
+		job.Spec.Resources = service.Spec.Resources
+	}
+}
+
+
+// ValidateJobInput validates a job before it was defaulted by BatchInferenceService values
+func ValidateJobInput(job api_types.InferenceJob) (errs []error) {
```

Comment:
Maybe I miss something here, but I suggest to combine functions `ValidateJobInput` and `ValidateJob` and call this function after filling possible defaults via `DefaultJob`.

Reply:
DefaultJob expects retrieved BatchInferenceService that correspond to creating BatchInferenceJob
Therefore we first need to validate that the user filled `Job.InferenceServiceID` in `ValidateJobInput `